cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -0,0 +1,221 @@
+		pod := pods.DefinePod(sriovnamespaces.Test)
+		pods.RedefineWithGuaranteedQoS(pod, "1", "100m")
+		pod = pods.RedefinePodWithNetwork(pod, "test-numa-0-exclude-topology-false")
+
+		pod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodFailed))
+			g.Expect(actualPod.Status.Reason).To(Equal("TopologyAffinityError"))
+		}).Should(Succeed())
+	})
+})
+
+func findDeviceOnNUMANode(node *corev1.Node, devices []*sriovv1.InterfaceExt, numaNode string) (*sriovv1.InterfaceExt, error) {
+	for _, device := range devices {
+		out, err := nodes.ExecCommandOnNode([]string{
+			"cat",
+			filepath.Clean(filepath.Join("/sys/class/net/", device.Name, "/device/numa_node")),
+		}, node)
+		if err != nil {
+			klog.Warningf("can't get device [%s] NUMA node: out(%s) err(%s)", device.Name, string(out), err.Error())
+			continue
+		}
+
+		if out == numaNode {
+			return device, nil
+		}
+	}
+
+	return nil, fmt.Errorf("can't find any SR-IOV device on NUMA [%s] for node [%s]. Available devices: %+v", numaNode, node.Name, devices)
+}
+
+func withNodeSelector(node *corev1.Node) func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork) {
+	return func(p *sriovv1.SriovNetworkNodePolicy, n *sriovv1.SriovNetwork) {
+		p.Spec.NodeSelector = map[string]string{
+			"kubernetes.io/hostname": node.Name,
+		}
+	}
+}
+
+func withPfNameSelector(pfNameSelector string) func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork) {
+	return func(p *sriovv1.SriovNetworkNodePolicy, n *sriovv1.SriovNetwork) {
+		p.Spec.NicSelector = sriovv1.SriovNetworkNicSelector{
+			PfNames: []string{pfNameSelector},
+		}
+		p.ObjectMeta.Name = "test-numa-test-policy-" + strings.ReplaceAll(pfNameSelector, "#", "-")
+	}
+}
+
+func withExcludeTopology(excludeTopology bool) func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork) {
+	return func(p *sriovv1.SriovNetworkNodePolicy, n *sriovv1.SriovNetwork) {
+		p.Spec.ExcludeTopology = excludeTopology
+	}
+}
+
+func withNumVFs(numVFs int) func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork) {
+	return func(p *sriovv1.SriovNetworkNodePolicy, n *sriovv1.SriovNetwork) {
+		p.Spec.NumVfs = numVFs
+	}
+}
+
+func withNetworkNameAndNamespace(namespace, name string) func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork) {
+	return func(p *sriovv1.SriovNetworkNodePolicy, n *sriovv1.SriovNetwork) {
+		n.ObjectMeta.Name = name
+		n.Spec.NetworkNamespace = namespace
+	}
+}
+
+func createSriovNetworkAndPolicy(opts ...func(*sriovv1.SriovNetworkNodePolicy, *sriovv1.SriovNetwork)) {
```

Comment:
What I don't like of this approach is, the modifier takes both the policy and the network.
We can split this in two (in the calling site) and have a more focused modifier for each side.